src/kernels/piecewisepolynomial.jl

```diff
@@ -0,0 +1,43 @@
+"""
+    PiecewisePolynomialKernel(P::AbstractMatrix)
+
+Piecewise Polynomial covariance function with compact support, v = 0,1,2,3.
+The kernel functions are 2v times continuously differentiable and the corresponding
+processes are hence v times  mean-square differentiable. The kernel function is:
+```math
+    κ(x,y) = max(1-r,0)^(j+v) * f(r,j) with j = floor(D/2)+v+1
+```
+where r is the Mahalanobis distance sqrt(maha(x,z)).
+
+"""
+struct PiecewisePolynomialKernel{T1<:Integer, T2<:Real, A<:AbstractMatrix{T2}} <: BaseKernel
+    v::T1
+    maha::A
+    function PiecewisePolynomialKernel(v::T1, maha::AbstractMatrix{T2}) where {T1<:Integer,T2<:Real}
+        if v ∉ [0,1,2,3] error("Invalid paramter v=$(v). Should be 0,1,2 or 3.") end
```

Comment:
Maybe `v` could be a type parameter? Then the branches in the function evaluations below would be optimized for the given value of `v` and be compiled away (i.e., instead of runtime checks the branching would happen at compile time).

Reply:
So are you suggesting we create types like `P0`, `P1`, `P2` and `P3` which is what we expect as a parameter `v`?

Reply:
No, I suggest defining
```suggestion
struct PiecewisePolynomialKernel{V,A<:AbstractMatrix{<:Real}} <: BaseKernel
```
where `V` will be 0, 1, 2, or 3.

Reply:
Oh okay. Makes sense.

Reply:
```julia
struct PiecewisePolynomialKernel{V, T2<:Real, A<:AbstractMatrix{T2}} <: BaseKernel
    v::V
    maha::A
    function PiecewisePolynomialKernel(v::T1, maha::AbstractMatrix{T2}) where {T1<:Integer,T2<:Real}
        if v ∉ [0,1,2,3] error("Invalid paramter v=$(v). Should be 0,1,2 or 3.") end
        LinearAlgebra.checksquare(maha)
        new{T1,T2,typeof(maha)}(v,maha)
    end
end
```
It this what you mean?